### PR TITLE
Add return Result<Value, Error> for paired returns

### PR DESCRIPTION
**Pull request by pavel-ship-it**
Each completion with return of (value, error) should have a completion with the return of Result type

### CODE REVIEW

File: RealmSwift/MongoClient.swift [hunk 1]
```diff
@@ -155,6 +155,21 @@ public typealias MongoUpdateBlock = (UpdateResult?, Error?) -> Void
 /// Block which returns the deleted Document on a successful delete operation, or an error should one occur.
 public typealias MongoDeleteBlock = RLMMongoDeleteBlock
 
+/// Block which returns Result.success(DocumentId) on a successful insert or Result.failure(error)
+public typealias MongoInsertResultBlock = (Result<AnyBSON?, Error>) -> Void
```
Comment (jsflax) [inline RealmSwift/MongoClient.swift:159]: Some of these result types should not be nullable.

**Comment by tgoyne** [comment]
I think it'd be better to replace the existing functions rather than add the Result version as an additional overload. Having both is mostly just confusing for people getting started with the API.

**Review by jsflax — COMMENTED** [review]
Agreed with Thomas.

File: RealmSwift/MongoClient.swift (hunk 1 above)
Comment (tgoyne) [reply RealmSwift/MongoClient.swift:159]: I think AnyBSON? is actually correct. BSON null is translated to `nil` and I think null is a legal document id?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@ x.y.z Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
 * Add the ability to listen for when a Watch Change Stream is opened when using Combine. Use `onOpen(event:)` directly after opening a `WatchPublisher` to register a callback to be invoked once the change stream is opened.
+* For paired return from completion handlers add return Result<Value, Error>
```
Comment (jsflax) [inline CHANGELOG.md:5]: This is a breaking change.

It also should start with an imperative:
```suggestion
* Change paired return types from Swift completion handlers to return `Result<Value, Error>`
```

File: RealmSwift/MongoClient.swift (hunk 1 above)
Comment (jsflax) [reply RealmSwift/MongoClient.swift:159]: Hmm... you are correct. I still find that to be a strange use case.

Comment (pavel-ship-it) [reply RealmSwift/MongoClient.swift:159]: There is a possible nullable successful result in some cases. I'll check it another time as it is not for all collection methods.

**Review by tgoyne — APPROVED** [review]

**Review by jsflax — APPROVED** [review]

**Pull request merged**